### PR TITLE
Feat/api connection

### PR DESCRIPTION
**Pull request by elton-dy**
implementation of sql lite for offline data management

### CODE REVIEW

**Review by sylvaindnd — DISMISSED** [review]

File: lib/screens/login_page.dart [hunk 1]
```diff
@@ -66,47 +66,47 @@ class _LoginPageState extends State<LoginPage> {
...
+              shape: RoundedRectangleBorder(
+                borderRadius: BorderRadius.circular(8.0),
+              ),
+            ),
+            child: const Text(
+              'Se connecter',
+              style: TextStyle(
+                color: Colors.white,
+                fontSize: 16.0,
+                fontWeight: FontWeight.bold,
```
Comment (FabienHP) [inline lib/screens/login_page.dart]:
```suggestion
              padding: const EdgeInsets.all(16.0),
              child: Container(
                decoration: BoxDecoration(
                  gradient: const LinearGradient(
                    colors: [
                      Color(0xFFFF0081),
                      Color(0xFFFF00E4),
                      Color(0xFFF15700),
                    ],
                    begin: Alignment.topLeft,
                    end: Alignment.bottomRight,
                  ),
                  borderRadius: BorderRadius.circular(8.0),
                ),
                child: ElevatedButton(
                  onPressed: () {
                    setState(() {
                      showScanner = true;
                    });
                    widget.qrScannerKey.currentState
                        ?.startScan(); // Action à effectuer lorsque le bouton est cliqué
                  },
                  style: ElevatedButton.styleFrom(
                    backgroundColor: Colors
                        .transparent, // Utilisez une couleur transparente pour le bouton
                    elevation: 0, // Supprimez l'ombre du bouton
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(8.0),
                    ),
                  ),
                  child: const Text(
                    'Se connecter',
                    style: TextStyle(
                      color: Colors.white,
                      fontSize: 16.0,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
```
Add whitespace for indentations

File: lib/screens/login_page.dart [hunk 2]
```diff
@@ -31,7 +31,7 @@ class _LoginPageState extends State<LoginPage> {
                 onDetect: (List<Barcode> barcodes, Uint8List? image) {
                   setState(() {
                     scannedQRCode =
-                        barcodes.isNotEmpty ? barcodes[0].rawValue : null;
+                    barcodes.isNotEmpty ? barcodes[0].rawValue : null;
```
Comment (FabienHP) [inline lib/screens/login_page.dart]:
```suggestion
                         barcodes.isNotEmpty ? barcodes[0].rawValue : null;
```
Add whitespace indentation

**Review by sylvaindnd — DISMISSED** [review]

**Review by FabienHP — CHANGES_REQUESTED** [review]
Fix identations and its ok for me

File: lib/screens/login_page.dart [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import 'dart:convert';
```
Comment (FabienHP) [inline lib/screens/login_page.dart]:
```suggestion

```
Delete not used import

**Review by FabienHP — CHANGES_REQUESTED** [review]

**Review by Lucas-Malatchoumy — COMMENTED** [review]
Séparer le code de database helper si on a le temps maybe

**Review by FabienHP — APPROVED** [review]
LGTM 👍

**Review by Lucas-Malatchoumy — APPROVED** [review]

**Review by FabienHP — APPROVED** [review]
LGTM 👍

**Review by sylvaindnd — APPROVED** [review]

**Pull request merged**